internal/contentmanager/contentmanager.go

```diff
@@ -2,222 +2,146 @@ package contentmanager
 	}
 
 	tgtLabels := labels.Set{
 		oclabels.OwnerKindKey: v1alpha1.ClusterExtensionKind,
 		oclabels.OwnerNameKey: ce.GetName(),
 	}
 
-	c, err := cache.New(cfg, cache.Options{
-		Scheme:               scheme,
-		DefaultLabelSelector: tgtLabels.AsSelector(),
-	})
-	if err != nil {
-		return fmt.Errorf("creating cache for ClusterExtension %q: %w", ce.Name, err)
+	dynamicSourcerer := &dynamicSourcerer{
+		// Due to the limitation outlined in the dynamic informer source,
+		// related to reusing an informer factory, we return a new informer
+		// factory every time to ensure we are not attempting to configure or
+		// start an already started informer
+		informerFactoryCreateFunc: func() dynamicinformer.DynamicSharedInformerFactory {
+			return dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicClient, time.Hour*10, metav1.NamespaceAll, func(lo *metav1.ListOptions) {
+				lo.LabelSelector = tgtLabels.AsSelector().String()
+			})
+		},
+		mapper: i.mapper,
 	}
+	cache = cmcache.NewCache(dynamicSourcerer, ce, i.syncTimeout)
+	i.caches[ce.Name] = cache
+	return cache, nil
+}
 
-	for _, obj := range objs {
-		err = ctrl.Watch(
-			source.Kind(
-				c,
-				obj,
-				handler.TypedEnqueueRequestForOwner[client.Object](
-					scheme,
-					i.mapper,
-					ce,
-					handler.OnlyControllerOwner(),
-				),
-				predicate.Funcs{
-					CreateFunc:  func(tce event.TypedCreateEvent[client.Object]) bool { return false },
-					UpdateFunc:  func(tue event.TypedUpdateEvent[client.Object]) bool { return true },
-					DeleteFunc:  func(tde event.TypedDeleteEvent[client.Object]) bool { return true },
-					GenericFunc: func(tge event.TypedGenericEvent[client.Object]) bool { return true },
-				},
-			),
-		)
-		if err != nil {
-			return fmt.Errorf("creating watch for ClusterExtension %q managed resource %s: %w", ce.Name, obj.GetObjectKind().GroupVersionKind(), err)
-		}
+// Delete stops and removes the Cache for the provided ClusterExtension
+func (i *managerImpl) Delete(ce *v1alpha1.ClusterExtension) error {
+	if ce == nil {
+		panic("nil ClusterExtension provided")
 	}
 
-	// TODO: Instead of stopping the existing cache and replacing it every time
-	// we should stop the informers that are no longer required
-	// and create any new ones as necessary. To keep the initial pass
-	// simple, we are going to keep this as is and optimize in a follow-up.
-	// Doing this in a follow-up gives us the opportunity to verify that this functions
-	// as expected when wired up in the ClusterExtension reconciler before going too deep
-	// in optimizations.
 	i.mu.Lock()
-	if extCache, ok := i.extensionCaches[ce.GetName()]; ok {
-		extCache.Cancel()
-	}
-
-	cacheCtx, cancel := context.WithCancel(context.Background())
-	i.extensionCaches[ce.Name] = extensionCacheData{
-		Cache:  c,
-		Cancel: cancel,
-	}
-	i.mu.Unlock()
-
-	go func() {
-		err := c.Start(cacheCtx)
+	defer i.mu.Unlock()
+	if cache, ok := i.caches[ce.Name]; ok {
+		err := cache.Close()
```

Comment:
Similar question that I asked earlier: is it possible to get some sort of "already closed" error here? If so, we should swallow it and proceed with removing the cache from the map.

Reply:
TLDR: No.

See: https://github.com/operator-framework/operator-controller/pull/1119#discussion_r1742728527